File: packages/script_entry_generator_rs/generator_test.rs
use prerender_metadata::{PrerenderMetadata, ScriptMetadata};
use generator::generate_entry_point;

#[test]
fn generate_entry_point_generates_from_metadata() {
    let metadata = PrerenderMetadata {
        scripts: vec![
            ScriptMetadata { path: String::from("foo/bar/baz.js") },
            ScriptMetadata { path: String::from("hello/world.js") },
        ],
    };

    let entry_point = generate_entry_point(metadata);

    assert_eq!(entry_point, "
import 'foo/bar/baz.js';
import 'hello/world.js';
    ".trim());
}

#[test]
fn generate_entry_point_generates_from_empty_metadata() {
    let metadata = PrerenderMetadata {
        scripts: vec![],
    };

    let entry_point = generate_entry_point(metadata);

    assert_eq!(entry_point, "");
}

File: tools/binaries/script_entry_generator/generator_test.ts
import { mockPrerenderMetadata } from '../../../common/models/prerender_metadata_mock';
import { generateEntryPoint } from './generator';

describe('generator', () => {
    describe('generateEntryPoint()', () => {
        it('generates an entry point', () => {
            const entryPoint = generateEntryPoint(mockPrerenderMetadata({
                scripts: [
                    { path: 'foo/bar/baz' },
                    { path: 'hello/world' },
                ],
            }), 2 /* importDepth */);
    
            expect(entryPoint).toBe(`
import '../../foo/bar/baz';
import '../../hello/world';
            `.trim());
        });

        it('generates an empty entry point when given no scripts', () => {
            const entryPoint = generateEntryPoint(mockPrerenderMetadata({
                scripts: [ /* no scripts */ ],
            }), 2 /* importDepth */);

            expect(entryPoint).toBe('');
        });

        it('generates a relative empty entry point with depth 0', () => {
            const entryPoint = generateEntryPoint(mockPrerenderMetadata({
                scripts: [
                    { path: 'foo/bar/baz' },
                    { path: 'hello/world' },
                ],
            }), 0 /* importDepth */);

            expect(entryPoint).toBe(`
import './foo/bar/baz';
import './hello/world';
            `.trim());
        });
    });
});
